esp-bootloader-esp-idf/src/ota.rs

```diff
@@ -215,76 +228,113 @@ where
 
     /// Sets the currently active OTA-slot.
     ///
-    /// Passing [Slot::None] will reset the OTA-data
-    pub fn set_current_slot(&mut self, slot: Slot) -> Result<(), crate::partitions::Error> {
-        if slot == Slot::None {
+    /// Passing [AppPartitionSubType::Factory] will reset the OTA-data.
+    ///
+    /// # Errors
+    ///
+    /// [Error::InvalidArgument] if [AppPartitionSubType::Test] is given or if the OTA app partition
+    /// number exceeds the value given to the constructor.
+    pub fn set_current_app_partition(&mut self, app: AppPartitionSubType) -> Result<(), Error> {
+        if app == AppPartitionSubType::Factory {
             self.flash.write(SLOT0_DATA_OFFSET, &[0xffu8; 0x20])?;
             self.flash.write(SLOT1_DATA_OFFSET, &[0xffu8; 0x20])?;
             return Ok(());
         }
 
-        let (seq0, seq1) = self.get_slot_seq()?;
+        if app == AppPartitionSubType::Test {
+            // cannot switch to the test partition - it's a partition
+            // which special built bootloaders will boot depending on the state of a pin
+            return Err(Error::InvalidArgument);
+        }
+
+        let idx = app as u8 - 0x10;
```

Comment:
I think some more comments in the function would help with clarity, because to my untrained eye, I don't really know what this is doing 😅 .